Add CRD support to k8s parser

tim-werner opened the pull request:
This will close issue [22445](https://github.com/pantsbuild/pants/issues/22445).
It is now possible to add a custom resource definition to the k8s-parser. 
```
[helm-k8s-parser]
crd = "pantsbuild/crd.py"
```

In the `crd.py` file a new CRD has to be defined according to the Hikaru documentation. The base class has to be named `CRD`.

tim-werner commented:
hi @cburroughs,
thank you very much for the feedback. I have linted the file and have added a brief release note (hopefully to the correct file). Please approve the workflow to run again.
Thx

alonsodomin commented:
yeah I like the idea, it's something I knew it would come up someday but avoided getting into that can of worms.

Regarding the list of CRDs, since you also need to know the class name, maybe a Hashmap is a better option.

tim-werner commented:
Hi @alonsodomin, @cburroughs ,

I have changed the code in such a way that it is now possible to register multiple crds. 

```
[helm-k8s-parser.crd]
"pantsbuild/crd.py"="CRD"
"pantsbuild/crd_cron2.py"="CRD2"
```
In addition I have added a section to the Helm Deployment documentation.

Please review.

Thx.

tim-werner commented:
I have openend a new pull request with the same name and the fixed conflicts. Maybe you can reopen this pull request as a admin. THx.

https://github.com/pantsbuild/pants/pull/22652
